wallet/src/service/Offers.ts

```diff
@@ -109,26 +141,34 @@ export const getOfferService = (
       return Object.fromEntries(entries);
     };
 
+    const readDisplayInfo = async (entry: PurseDisplayInfo) => {
+      mustMatch(harden(entry.amount), CapDataShape, 'amount');
```

Comment:
Why enforce the shape here instead of deferring to the marshaller what it will accept?